src/http.cc

```diff
@@ -1933,6 +2071,49 @@ HttpStateData::httpBuildRequestHeader(HttpRequest * request,
     strConnection.clean();
 }
 
+void
+HttpStateData::makeUpgradeHeaders(HttpHeader &hdr_out)
+{
+    // We need virgin request received by squid. This is stored to
+    // AccessLogEntry::request member.
+    const HttpHeader &hdr_in = fwd->al->request ? fwd->al->request->header :request->header;
+
+    if (!hdr_in.has(Http::HdrType::UPGRADE))
+        return;
+
+    const String upgradeIn = hdr_in.getList(Http::HdrType::UPGRADE);
```

Comment:
I do not think but If you insist we can do it. 

However in most cases the Upgrade header does not exist, so as is the code is faster because an unneeded String is not constructed in these cases(actually two String objects are build for HttpHeader::getList(Http::HdrType) return empty string)

Reply:
It is not an optimisation, it is required. We can not just remove it for example, even if currently this is will not cause problems.

 It can be replaced with a different check (the `if (!upgradeIn.size()) return;`) or can be moved out of this method, but still required because we are not checking if this header exist before call this method. Someone may concludes that we are doing if we remove it.